### ISSUE TITLE
add local geth to github action

### ISSUE THREAD

**Issue by Vid201** [Vid201/aa-bundler#60]
Some tests will rely on local geth node, would be useful to have this in github action

**Comment by zsluedem** [comment]
Currently, each test would start a geth node by itself. Do you mean start only one and put it in the backgroud?

**Comment by Vid201** [comment]
Yes, that was the idea. A lot of unit tests also need a geth client since several checks depend on that (e.g., https://github.com/Vid201/aa-bundler/blob/main/src/uopool/services/sanity_check.rs#L356).

We could run geth in the background, deploy all necessary contracts (and possibly do other things) before running tests.

**Comment by zsluedem** [comment]
I used to have problems with one big Geth dev mode because the rust tests are running in parallel. Then `nonce` is a little bit tricky.

**Comment by Vid201** [comment]
Hmm, you're right. Ok, then let's leave this for now.

**Comment by TobieTom** [comment]
Can I handle this issue? I currently just learned Rust and Cairo and am looking to add to my experience since I am new to the space
